pkgs/applications/audio/bambootracker/default.nix

```diff
@@ -0,0 +1,54 @@
+{ mkDerivation, stdenv, fetchFromGitHub, installShellFiles
+, qmake, qtbase, qtmultimedia, qttools
+, alsaSupport ? stdenv.hostPlatform.isLinux, alsaLib
+, pulseSupport ? stdenv.hostPlatform.isLinux, libpulseaudio
+, jackSupport ? stdenv.hostPlatform.isUnix, libjack2
+}:
+
+let
+
+  opt = stdenv.lib.optional;
+
+in mkDerivation rec {
+  pname = "bambootracker";
+  version = "0.3.5";
+
+  src = fetchFromGitHub {
+    owner = "rerrahkr";
+    repo = "BambooTracker";
+    rev = "v${version}";
+    sha256 = "06ij0k5sfns1zakl6qbxr5j91dj1kfj3s5jc3mpqzxc3zpc3avkv";
+  };
+
+  sourceRoot = "source/BambooTracker";
+
+  nativeBuildInputs = [ qmake qttools installShellFiles ];
+
+  buildInputs = [ qtbase qtmultimedia ]
+  ++ opt alsaSupport alsaLib
+  ++ opt pulseSupport libpulseaudio
+  ++ opt jackSupport libjack2;
+
+  qmakeFlags = []
+  ++ opt alsaSupport  "DEFINES+=__LINUX_ALSA__"
+  ++ opt pulseSupport "DEFINES+=__LINUX_PULSE__"
+  ++ opt jackSupport  "DEFINES+=__UNIX_JACK__";
+
+  postInstall = ''
+    install -Dm644 ../BambooTracker.desktop $out/share/applications/BambooTracker.desktop
+    installManPage ../BambooTracker*.1
+    cp -r ../demos $out/share/BambooTracker/
```

Comment:
imo keeping the `demos` subdirectory (`cp -r ../demos $out/share/BambooTracker/demos`) would be a bit nicer than having its contents copied directly in `share/BamboTracker`, but this is fine too

Reply:
Whoops, that was my intention. Fixed.